google-cloud-spanner/src/main/java/com/google/cloud/spanner/TraceUtil.java

```diff
@@ -54,6 +54,16 @@
         "Status", AttributeValue.stringAttributeValue(e.getErrorCode().toString()));
   }
 
+  static void setWithFailure(Span span, Throwable e) {
+    if (e instanceof SpannerException) {
+      span.setStatus(
+          StatusConverter.fromGrpcStatus(((SpannerException) e).getErrorCode().getGrpcStatus())
+              .withDescription(e.getMessage()));
+    } else {
+      span.setStatus(Status.INTERNAL.withDescription(e.getMessage()));
+    }
+  }
+
   static void endSpanWithFailure(Span span, Throwable e) {
```

Comment:
is endSpanWithFailure used anywhere now? If not then please remove it.

Reply:
This has been used in a few places (ex: [singleUse](https://github.com/googleapis/java-spanner/blob/master/google-cloud-spanner/src/main/java/com/google/cloud/spanner/DatabaseClientImpl.java#L113)) where span does not end except when there is a failure.

See: https://github.com/googleapis/google-cloud-java/pull/2677/files#r157323309 for original reason.